### ISSUE TITLE
CWL failure when passing an array with default values

### ISSUE THREAD

**Issue by arostamianfar** [DataBiosphere/toil#2814]
It looks like PR #2785 introduced a bug that throws the following exception at runtime for particular inputs:

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/toil/worker.py", line 346, in workerScript
    job._runner(jobGraph=jobGraph, jobStore=jobStore, fileStore=fileStore, defer=defer)
  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 1392, in _runner
    returnValues = self._run(jobGraph, fileStore)
  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 1329, in _run
    return self.run(fileStore)
  File "/usr/local/lib/python2.7/dist-packages/toil/cwl/cwltoil.py", line 952, in run
    jobobj[key][0]] is None:
TypeError: tuple indices must be integers, not unicode
```

CWL input to reproduce:

```
cwlVersion: v1.0
class: Workflow
requirements:
  SubworkflowFeatureRequirement: {}
inputs:
  strings:
    type: string[]
    default: ['hello', 'world']
steps:
  step1:
    run:
      class: Workflow
      inputs:
        strings: string[]
      outputs: []
      steps:
        step_inline:
          in:
            strings: strings
          out: []
          run:
            class: CommandLineTool
            baseCommand: [echo]
            outputs: []
            inputs:
              strings:
                default: ['hello']
                type: string[]
                inputBinding:
                  position: 1
    in:
      strings: strings
    out: []
outputs: []
```

I've verified that the above input works fine when the PR is reverted. It also works fine if `default: ['hello']` is removed from the CWL file.

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-432)
┆Issue Number: TOIL-432

**Comment by arostamianfar** [comment]
@mr-c thanks for the fix, however, the same error is still happening after rebasing to master:
```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/toil/worker.py", line 366, in workerScript
    job._runner(jobGraph=jobGraph, jobStore=jobStore, fileStore=fileStore, defer=defer)
  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 1392, in _runner
    returnValues = self._run(jobGraph, fileStore)
  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 1329, in _run
    return self.run(fileStore)
  File "/usr/local/lib/python2.7/dist-packages/toil/cwl/cwltoil.py", line 954, in run
    jobobj[key][0]] is None):
TypeError: tuple indices must be integers, not unicode
```
Just ran with `toil-cwl-runner test.cwl empty.yaml` using the CWL text above.

**Comment by arostamianfar** [comment]
Will be fixed by #2845 .. thanks @kaushik-work!
